IdentityCore/src/workplacejoin/mac/MSIDWorkPlaceJoinUtil.m

```diff
@@ -158,4 +197,65 @@ + (SecIdentityRef)copyWPJIdentity:(__unused id<MSIDRequestContext>)context
     return identityRef; //Caller must call CFRelease
 }
 
+#pragma mark - Lookup by private key
+
++ (MSIDRegistrationInformation *)findDefaultWPJRegistrationInfoWithContext:(id<MSIDRequestContext>)context
+{
+    OSStatus status = noErr;
+    CFTypeRef privateKeyCFDict = NULL;
+    
+    // Set the private key query dictionary.
+    NSDictionary *queryCert = @{ (__bridge id)kSecClass : (__bridge id)kSecClassKey,
+                                 (__bridge id)kSecAttrApplicationTag: kMSIDPrivateKeyIdentifier,
+                                 (__bridge id)kSecReturnAttributes: @YES,
+                                 (__bridge id)kSecReturnRef: @YES};
+    
+    // Get the certificate. If the certificate is not found, this is not considered an error.
+    status = SecItemCopyMatching((__bridge CFDictionaryRef)queryCert, (CFTypeRef*)&privateKeyCFDict);
+    if (status != errSecSuccess)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to find workplace join private key with status %ld", (long)status);
+        return nil;
+    }
+        
+    NSDictionary *privateKeyDict = CFBridgingRelease(privateKeyCFDict);
+    
+    /*
+     kSecAttrApplicationLabel
+     For asymmetric keys this holds the public key hash which allows digital identity formation (to form a digital identity, this value must match the kSecAttrPublicKeyHash ('pkhh') attribute of the certificate)
+     */
+    NSData *applicationLabel = privateKeyDict[(__bridge id)kSecAttrApplicationLabel];
+
+    if (!applicationLabel)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Unexpected key found without application label. Aborting lookup");
+        return nil;
+    }
+    
+    SecKeyRef privateKeyRef = (SecKeyRef)CFBridgingRetain(privateKeyDict[(__bridge id)kSecValueRef]);
+    
+    if (!privateKeyRef)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"No private key ref found. Aborting lookup.");
+        return nil;
+    }
+        
+    NSDictionary *certQuery = @{(__bridge id)kSecClass : (__bridge id)kSecClassCertificate,
+                                (__bridge id)kSecAttrPublicKeyHash: applicationLabel};
+    
+    SecCertificateRef certRef;
+    status = SecItemCopyMatching((__bridge CFDictionaryRef)certQuery, (CFTypeRef*)&certRef);
+    
+    if (status != errSecSuccess)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to find certificate for public key hash with status %ld", (long)status);
+        return nil;
+    }
+    
+    return [self registrationInformationWithIdentity:nil
```

Comment:
It it possible to not intentionally pass nil and still achieve the differentiation between findDefaultWPJRegistrationInfoWithContext or findWPJRegistrationInfoWithAuthorities.

Reply:
since it's optional now, it doesn't have to be on the initializer. I think I'll remove it from initializer altogether. What do you think?